server.go

```diff
@@ -102,7 +106,16 @@ func NewServer() *Server {
 	config := NewServerConfig(cfgPath)
 
 	log.Debug("Configuration loaded with public key: ", config.PublicKey)
-	assets := http.FileServer(&assetfs.AssetFS{Asset: Asset, AssetDir: AssetDir, AssetInfo: AssetInfo, Prefix: ""})
+
+	var fsys fs.FS = assetsFS
+	if f, err := fs.Sub(fsys, "ui/dist"); err != nil {
+		log.Error(fmt.Errorf("ui/dist does not exist in fs :%w", err))
```

Comment:
This will never execute as the program as specified now because go:embed ui/dist fill fail the compilation if it does not exist. Just error printing the error to be safe is something changes.